core/shared/src/main/scala/scalaz/zio/internal/Scheduler.scala

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2018 - 2019 John A. De Goes. All rights reserved.
+package scalaz.zio.internal
+
+import scala.concurrent.duration._
+import java.util.concurrent._
+import java.util.concurrent.atomic.AtomicInteger
+
+trait Scheduler {
+  import Scheduler.CancelToken
+
+  def schedule(executor: Executor, task: Runnable, duration: Duration): CancelToken
```

Comment:
other schedule method:
schedulerOnce
scheduleAtFixRate
initial delay etc,
where the task execution should be taken

Reply:
You can do that in `IO` and it's a billion times easier (`Schedule`, `IO#repeat`, etc.).